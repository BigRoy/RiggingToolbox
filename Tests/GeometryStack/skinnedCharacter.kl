
require RiggingToolbox;

inline __compile_Vec3Attribute_copyFrom() { Vec3 a[]; Vec3 b[]; Vec3Attribute_copyFrom<<<1@true>>>(a, b); }
inline __compile_SkinningAttribute_copyFrom() { SkinningAttributeDataUnion a; SkinningAttributeDataUnion b; SkinningAttribute_copyFrom<<<1@true>>>(a, b); }
inline __compile_skinningModifier_skinMeshPositions() { 
  PolygonMeshDataUnion a; Vec3 b<>; SkinningAttributeDataUnion d; Mat44 e[]; skinningModifier_skinMeshPositions<<<1@true>>>(a, b, d, e); 
}
inline __compile_skinningModifier_skinMeshPositionsAndNormals() { 
  PolygonMeshDataUnion a; Vec3 b<>; Vec3 c<>; SkinningAttributeDataUnion d; Mat44 e[]; skinningModifier_skinMeshPositionsAndNormals<<<1@true>>>(a, b, c, d, e); 
}


operator entry(){

  StartFabricProfiling();

  GeometryStack stack();

  AlembicSkinnedMeshGeometryGenerator alembicGenerator();
  SkinningModifier skiningModifier();
  stack.addGeometryOperator(alembicGenerator);
  stack.addGeometryOperator(skiningModifier);

  alembicGenerator.setFilePath("${FABRIC_RIGGINGTOOLBOX_PATH}/Tests/GeometryStack/Resources/skinnedTube.abc");

  String geometryNames[];
  geometryNames.push("pCylinderShape1");
  alembicGenerator.setGeometryNames(geometryNames);

  EvalContext context();
  GeometrySet geomSet = stack.evaluate(context);

  // Modify the pose and then reevaluate. 
  Mat44 pose[];
  pose.resize(2);
  pose[0] = Xfo(Vec3(3, 4, 5)).toMat44();
  pose[1] = Xfo(Vec3(10, 20, 5)).toMat44();
  skiningModifier.setPose(pose);

  stack.evaluate(context);

  StopFabricProfiling();

  report( GetEvalPathReport() );
}

