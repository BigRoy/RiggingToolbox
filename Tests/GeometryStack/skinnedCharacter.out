Importing:D:\Projects\RiggingToolbox/Tests/GeometryStack/Resources\skinnedTube.abc
AlembicGeometryAttributesReader: Property "EnvelopeWeights" not supported.
AlembicGeometryAttributesReader: Property "MappingIndices" not supported.
AlembicGeometryAttributesReader: Property "EnvelopeDeformerIndices" not supported.
AlembicGeometryAttributesReader: Property "EnvelopeDeformers" not supported.
function GeometrySet GeometryStack.evaluate!(EvalContext)
--function AlembicSkinnedMeshGeometryGenerator.evaluate!(EvalContext, io GeometrySet)
--function GeometrySet.updateCache!(UInt32, GeometryOperator):1
----function GeometryAttributeCache.update!(Geometry[]):SkinningModifier:["positions","normals","skinningData"]
--function SkinningModifier.evaluate!(EvalContext, io GeometrySet)
--function GeometrySet.updateCache!(UInt32, GeometryOperator):2
----function GeometryAttributeCache.update!(Geometry[]):ComputeNormalsModifier:["positions","normals"]
--function ComputeNormalsModifier.evaluate!(EvalContext, io GeometrySet)
--function GeometrySet.updateCache!(UInt32, GeometryOperator):3
----function GeometryAttributeCache.update!(Geometry[]):ComputeTangentsModifier:["positions","uvs0","tangents"]
--function ComputeTangentsModifier.evaluate!(EvalContext, io GeometrySet)
function SkinningModifier.setPose!(Mat44[])
--function GeometryStack.notify!(Notifier, String, String):SkinningModifier.changed
----function GeometrySet.rewind!(UInt32):1
------function GeometryAttributeCache.restore?(io Geometry[]):ComputeTangentsModifier:["positions","uvs0","tangents"]
------function GeometryAttributeCache.restore?(io Geometry[]):ComputeNormalsModifier:["positions","normals"]
------function GeometryAttributeCache.restore?(io Geometry[]):SkinningModifier:["positions","normals","skinningData"]
function GeometrySet GeometryStack.evaluate!(EvalContext)
--function SkinningModifier.evaluate!(EvalContext, io GeometrySet)
--function GeometrySet.updateCache!(UInt32, GeometryOperator):2
----function GeometryAttributeCache.update!(Geometry[]):ComputeNormalsModifier:["positions","normals"]
--function ComputeNormalsModifier.evaluate!(EvalContext, io GeometrySet)
--function GeometrySet.updateCache!(UInt32, GeometryOperator):3
----function GeometryAttributeCache.update!(Geometry[]):ComputeTangentsModifier:["positions","uvs0","tangents"]
--function ComputeTangentsModifier.evaluate!(EvalContext, io GeometrySet)
function SkinningModifier.setPose!(Mat44[])
--function GeometryStack.notify!(Notifier, String, String):SkinningModifier.changed
----function GeometrySet.rewind!(UInt32):1
------function GeometryAttributeCache.restore?(io Geometry[]):ComputeTangentsModifier:["positions","uvs0","tangents"]
------function GeometryAttributeCache.restore?(io Geometry[]):ComputeNormalsModifier:["positions","normals"]
------function GeometryAttributeCache.restore?(io Geometry[]):SkinningModifier:["positions","normals","skinningData"]
function GeometrySet GeometryStack.evaluate!(EvalContext)
--function SkinningModifier.evaluate!(EvalContext, io GeometrySet)
--function GeometrySet.updateCache!(UInt32, GeometryOperator):2
----function GeometryAttributeCache.update!(Geometry[]):ComputeNormalsModifier:["positions","normals"]
--function ComputeNormalsModifier.evaluate!(EvalContext, io GeometrySet)
--function GeometrySet.updateCache!(UInt32, GeometryOperator):3
----function GeometryAttributeCache.update!(Geometry[]):ComputeTangentsModifier:["positions","uvs0","tangents"]
--function ComputeTangentsModifier.evaluate!(EvalContext, io GeometrySet)

